pkg/server/forward.go

```diff
@@ -121,38 +129,24 @@ func (s *Store) WriteMetrics(ctx context.Context, p *store.PartitionedMetrics) e
 
 		meanDrift := timeseriesMeanDrift(timeseries, time.Now().Unix())
 		if math.Abs(meanDrift) > 10 {
-			level.Info(s.logger).Log("msg", "mean drift from now for clusters", "partitionkey", p.PartitionKey, "drift", fmt.Sprintf("%.3fs", meanDrift))
+			level.Info(logger).Log("msg", "mean drift from now for clusters", "clusterID", clusterID, "drift", fmt.Sprintf("%.3fs", meanDrift))
 		}
 
 		if resp.StatusCode/100 != 2 {
-			return fmt.Errorf("response status code is %s", resp.Status)
+			// surfacing upstreams error to our users too
+			http.Error(w, fmt.Errorf("response status code is %s", resp.Status).Error(), resp.StatusCode)
```

Comment:
This isn't correct IMO. If telemeter-server gets a 400 from the upstream then it is not the user's fault that there is an auth error, but rather an internal server error. The user should not receive a 400 for an improperly configured server. Instead the user should get a 500.

Reply:
Right. That makes sense. Keep in mind that this is the request send from Telemeter to Thanos Receivers. I guess it still holds true then too?